src/frontend/org/voltdb/calciteadapter/rules/inlining/VoltDBPLimitScanMergeRule.java

```diff
@@ -21,24 +21,55 @@
 import org.apache.calcite.plan.RelOptRuleCall;
 import org.apache.calcite.rel.RelNode;
 import org.voltdb.calciteadapter.rel.physical.AbstractVoltDBPTableScan;
+import org.voltdb.calciteadapter.rel.physical.VoltDBPCalc;
 import org.voltdb.calciteadapter.rel.physical.VoltDBPLimit;
+import org.voltdb.calciteadapter.rel.physical.VoltDBPSerialAggregate;
 
 public class VoltDBPLimitScanMergeRule extends RelOptRule {
 
-    public static final VoltDBPLimitScanMergeRule INSTANCE = new VoltDBPLimitScanMergeRule();
+    public static final VoltDBPLimitScanMergeRule INSTANCE_1 = new VoltDBPLimitScanMergeRule(false);
+    public static final VoltDBPLimitScanMergeRule INSTANCE_2 = new VoltDBPLimitScanMergeRule(true);
```

Comment:
I was just thinking if creating the patterns directly here and sending as operand to the constructor instead of a flag makes it cleaner.

Reply:
Indeed. Updated